test-gen-2/src/LightSm.csx

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env dotnet-script
+// This is a c# script file
+
+#r "nuget: StateSmith, 0.9.12-alpha" // this line specifies which version of StateSmith to use and download from c# nuget web service.
+
+using StateSmith.Input.Expansions;
+using StateSmith.Output.UserConfig;
+using StateSmith.Runner;
+using StateSmith.SmGraph;  // Note using! This is required to access StateMachine and NamedVertex classes...
+using StateSmith.SmGraph.Visitors;
+using StateSmith.Common;
+
+TextWriter mermaidCodeWriter = new StringWriter();
+SmRunner htmlRunner = new(diagramPath: "LightSm.drawio.svg", new LightSmRenderConfig(), transpilerId: TranspilerId.JavaScript);
+htmlRunner.SmTransformer.InsertBeforeFirstMatch(
+    StandardSmTransformer.TransformationId.Standard_FinalValidation,
+    new TransformationStep(id: "some string id", action: (sm) =>
+    {
+        var visitor = new MermaidGenerator(mermaidCodeWriter);
+        sm.Accept(visitor);
+        visitor.Print(); // print the mermaid code to the mermaidcodewriter
+        List<string> events = GetNonDoEvents(sm);
+        events = events.ConvertAll(e => e.ToUpper()); // TODO what is the right way to get the event name in the proper case?
```

Comment:
We can easily get the events from the js state machine: https://github.com/StateSmith/exp-test-gen/blob/6b4fe3a9d69ef599280a7af400d9cec06a7f7504/src2/LightSm2Sim/index.html#L12-L21

Reply:
okay that is 1000x better